### PR TITLE
Regex search tables

### LINKED ISSUE #2534: Table view search would be more useful if there was a regular expression implementation
User feedback would appreciate the ability to specify value >50 or !string or some other form of search within table views. Exports of the view would then be limited by the search, allowing simple analysis.

### PR DESCRIPTION
**Pull request by deeptailor**
Fixes #2534 

## Author Checklist

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? N/A
- Command line build passes? Yes
- Changes have been smoke-tested? Yes

Styling done and approved by @charlesh88

### CODE REVIEW

**Review by charlesh88 — CHANGES_REQUESTED** [review]
Running locally with a SWG in a Telemetry Table, I'm finding that after I enable regex and click in the input, it's not displaying the first character I type - but the field does think there's something in it.
Screen Shot 2020-04-21 at 5 07 29 PM

So I have to type "/" twice to get one to appear - or click on an active thing outside the input, like another regex button or even the Inspector splitter. I'm also seeing weird stuff with the Backspace key: putting the cursor to the right of what looks like one "/" and hitting Backspace doesn't work - but backspacing again does work.

**Comment by deeptailor** [comment]
@charlesh88 - thanks for catching this. I fixed the issue you describe

**Comment by deeptailor** [comment]
@charlesh88 - I fixed the issue you described, could you approve the changes?

File: src/plugins/telemetryTable/components/table.vue [hunk 1]
```diff
@@ -869,6 +889,18 @@ export default {
...
+            this.$set(this.filters, key, '');
+
+            if (this.enableRegexSearch[key] === undefined) {
+                this.$set(this.enableRegexSearch, key, true)
+            } else {
+                this.$set(this.enableRegexSearch, key, !this.enableRegexSearch[key]);
+            }
+        },
+        isCompleteRegex(string) {
+            return (string.length > 2 && string[0] === '/' && string[string.length - 1] === '/')
```
Comment (akhenry) [inline src/plugins/telemetryTable/components/table.vue:903]: 👍  Hehe, I'd have used a regex to test if it was a regex, but this approach is way more performant 😄

**Comment by davetsay** [comment]
Looks good. I have a couple comments and questions.

- / \\ / results in console error (forward slash, back slash, forward slash)
```
FilteredTableRowCollection.js?25a5:64 Uncaught SyntaxError: Invalid regular expression: /swg \/: \ at end of pattern
    at new RegExp (<anonymous>)
    at FilteredTableRowCollection.setColumnRegexFilter (FilteredTableRowCollection.js?25a5:64)
    at VueComponent.filterChanged (table.vue?5cb1:559)
    at Proxy.boundFn (vue.js?ba4c:194)
    at complete (index.js?45e9:7697)
    at delayed (index.js?45e9:7707)
```

- filter breaks if i 1) enter a valid filter (non-regex), 2) add a character to match nothing, 3) remove the character in step 2. I'm unsure if this is a known issue.
cannot unfilter

- is it possible to have the user click the regex button XOR type the open and close forward slashes? I understand the slashes are required to identify a valid regex expression. Obviously it would take more work but using vscode as an example - regex search does not require open close slashes and will alert the user for malformed regex expressions.
image

**Review by akhenry — CHANGES_REQUESTED** [review]
Approach looks good to me. We need some tests though.

**Review by davetsay — APPROVED** [review]
## Reviewer Checklist
| | |
| --- | :---: |
| Changes appear to address issue? | Y |
| Appropriate unit tests included? | N/A |
| Code style and in-line documentation are appropriate? | Y |
| Commit messages meet standards? | Y |

**Review by davetsay — DISMISSED** [review]

**Review by charlesh88 — DISMISSED** [review]
LGTM

**Comment by deeptailor** [comment]
@akhenry - Fixed the conflict

**Comment by akhenry** [comment]
@deeptailor Just waiting for tests in order to merge this into master.

**Comment by deeptailor** [comment]
@akhenry - Sorry for the delay on this branch, i added some tests and should be good to go

**Review by akhenry — APPROVED** [review]

**Comment by akhenry** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue)

**Comment by RadotJar** [comment]
>     * is it possible to have the user click the regex button XOR type the open and close forward slashes? I understand the slashes are required to identify a valid regex expression. Obviously it would take more work but using vscode as an example - regex search does not require open close slashes and will alert the user for malformed regex expressions.

Was there any followup on this? Would be a nice improvement to user experience if one didn't have to write the slashes at start and end

**Pull request merged**